main.c

```diff
@@ -15,14 +15,47 @@ how to use the page table and disk interfaces.
 #include <string.h>
 #include <errno.h>
 
-void page_fault_handler( struct page_table *pt, int page )
-{
-	printf("page fault on page #%d\n",page);
-	exit(1);
+struct disk *disk;
+
+// My understanding is that, when a page fault occurs, we need to evict data from a page.
+// To do that, we must preserve memory writes by writing the frame where the fault occurs
+// to the disk. Then, the next time the frame is loaded, the data will be preserved.
+void page_fault_handler( struct page_table *pt, int page ) {
+	int frame;
+	for(int f=0; f < pt->nframes; f++){
+
+
+		// Find a frame that is not already allocated to a page.
+		int page;
+		for (page=0; page<pt->npages; page++){
+			if(pt->page_mapping[page] == f){ /* && page_bits[page] != 0 */ // We could also check the page bits, but it doesn't matter too much.
+				break;
```

Comment:
I think we need to check page_bits[page] too because initially all the page_mappings are going to be equal to 0 so technically every page is mapped to the first frame.